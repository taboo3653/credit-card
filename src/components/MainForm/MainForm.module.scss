.MainForm {
  background-color: #fff;
  padding: 35px;
  box-shadow: 0 30px 60px 0 rgba(90, 116, 148, 0.4);
  border-radius: 10px;
  padding-top: 180px;
}
