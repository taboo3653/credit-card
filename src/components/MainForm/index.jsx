import React from 'react';
import { Formik } from 'formik';

import Form from 'react-bootstrap/Form';
import InputGroup from 'react-bootstrap/InputGroup';
import Button from 'components/Button';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';

class MainForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Form noValidate>
        <Form.Group>
          <Form.Label>Card Number</Form.Label>
          <Form.Control type="text" name="cardNumber" />
          <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
        </Form.Group>

        <Form.Group>
          <Form.Label>Card Holder</Form.Label>
          <Form.Control type="text" name="cardHolder" />
          <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
        </Form.Group>

        <Form.Row>
          <Form.Group as={Col} md="8" controlId="validationFormik01">
            <Form.Label>First name</Form.Label>
            <Form.Row>
              <Col>
                <Form.Control type="text" name="firstName" />
              </Col>
              <Col>
                <Form.Control type="text" name="firstName" />
              </Col>
            </Form.Row>
          </Form.Group>

          <Form.Group
            as={Col}
            md="4"
            controlId="validationFormikUsername"
          >
            <Form.Label>Last name</Form.Label>
            <Form.Control type="text" name="lastName" />
            <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
          </Form.Group>
        </Form.Row>
        <Button type="submit">Submit form</Button>
      </Form>
    );
  }
}

export default MainForm;
