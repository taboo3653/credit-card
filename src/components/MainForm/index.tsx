import * as React from 'react';

import Form from 'react-bootstrap/Form';
import Col from 'react-bootstrap/Col';

import Button from 'components/Button';
import Label from 'components/Label';
import Input from 'components/Inputs/Input';
import CardNumberInput from 'components/Inputs/CardNumberInput';

import styles from './MainForm.module.scss';

export interface Props {
  onChange: (event: React.ChangeEvent<unknown>) => void;
  numberValue: string;
}

function MainForm({
  numberValue,
  onChange: handleChange,
}: Props): React.ReactElement {
  return (
    <Form className={styles.MainForm}>
      <Form.Group>
        <Label>Card Number</Label>
        <CardNumberInput
          name="number"
          value={numberValue}
          onChange={handleChange}
        />
        <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
      </Form.Group>

      <Form.Group>
        <Label>Card Holder</Label>
        <Input name="cardHolder" />
        <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
      </Form.Group>

      <Form.Row>
        <Form.Group
          as={Col}
          xs="8"
          className="pr-4"
          controlId="validationFormik01"
        >
          <Label>Expiration Date</Label>
          <Form.Row>
            <Col>
              <Input name="firstName" placeholder="Month" />
            </Col>
            <Col>
              <Input name="firstName" placeholder="Year" />
            </Col>
          </Form.Row>
        </Form.Group>

        <Form.Group
          as={Col}
          xs="4"
          controlId="validationFormikUsername"
        >
          <Label>CVV</Label>
          <Input name="lastName" />
          <Form.Control.Feedback>Looks good!</Form.Control.Feedback>
        </Form.Group>
      </Form.Row>
      <Button type="submit">Submit</Button>
    </Form>
  );
}

export default MainForm;
