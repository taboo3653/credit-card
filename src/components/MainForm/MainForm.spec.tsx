import * as React from 'react';
import { shallow } from 'enzyme';

import MainForm, { Props } from '.';

const props: Props = {
  onChange: jest.fn(),
  numberValue: '',
};

it('snapshot', () => {
  const component = shallow(<MainForm {...props} />);
  expect(component).toMatchSnapshot();
});
