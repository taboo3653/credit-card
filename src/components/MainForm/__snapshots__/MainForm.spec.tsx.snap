// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<Form
  className="MainForm"
  inline={false}
>
  <FormGroup>
    <Label>
      Card Number
    </Label>
    <Input
      name="number"
      onChange={[MockFunction]}
      value=""
    />
    <Feedback
      type="valid"
    >
      Looks good!
    </Feedback>
  </FormGroup>
  <FormGroup>
    <Label>
      Card Holder
    </Label>
    <Input
      name="cardHolder"
    />
    <Feedback
      type="valid"
    >
      Looks good!
    </Feedback>
  </FormGroup>
  <FormRow>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Col",
          "render": [Function],
        }
      }
      className="pr-4"
      controlId="validationFormik01"
      xs="8"
    >
      <Label>
        Expiration Date
      </Label>
      <FormRow>
        <Col>
          <Input
            name="firstName"
            placeholder="Month"
          />
        </Col>
        <Col>
          <Input
            name="firstName"
            placeholder="Year"
          />
        </Col>
      </FormRow>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Col",
          "render": [Function],
        }
      }
      controlId="validationFormikUsername"
      xs="4"
    >
      <Label>
        CVV
      </Label>
      <Input
        name="lastName"
      />
      <Feedback
        type="valid"
      >
        Looks good!
      </Feedback>
    </FormGroup>
  </FormRow>
  <Button
    type="submit"
  >
    Submit
  </Button>
</Form>
`;
